### PR TITLE
Forwarder use agent name in folder

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR allows several agents to store payloads on disk.

- Use one folder by Agent kind (`core` for the core Agent)
- Make sure the folder exists when calling `computeMaxStorage` (If `forwarder_storage_path` is set, `maxStorage.computeMaxStorage(0)` failed because the folder may not exist yet.)
- Restrict the forwarder folder access to the current user / group (On Windows, settings `forwarder_storage_path` to a value outside `C:\ProgramData\Datadog` makes the folder available to all users.)
- Fix a typo for `forwarder_storage_max_disk_ratio` in config_template.yaml

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan
**QA should be done on Windows and Unix system.**

For `Use one folder by Agent kind (core for the core Agent)` see https://github.com/DataDog/datadog-agent/pull/7060

For `Make sure the folder exists when calling computeMaxStorage` set  `forwarder_storage_path` to a not existing path and check storage on disk is enabled.

For `Restrict the forwarder folder access to the current user / group`, when setting `forwarder_storage_path` checks `.retry` files are not accessible to a no admin user.

### CODE REVIEW

File: pkg/forwarder/transactions_file_storage.go [hunk 1]
```diff
@@ -50,7 +50,12 @@ func newTransactionsFileStorage(
 	if err := storage.reloadExistingRetryFiles(); err != nil {
 		return nil, err
 	}
-	return storage, nil
+
+	// Check if there is an error when computing the available space
+	// in this function to warn the user sonner (and not when there is an outage)
```
Comment (ruthnaebeck) [inline pkg/forwarder/transactions_file_storage.go:55]:
```suggestion
	// in this function to warn the user sooner (and not when there is an outage)
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Docs review

File: pkg/util/filesystem/permission_nowindows.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+// Create a new instance of `Permission`
+func NewPermission() (*Permission, error) {
+	return &Permission{}, nil
+}
+
+// Set the permission of `path` to the current user and current group.
+func (p *Permission) SetPermToCurrentUserAndGroup(path string) error {
+	return os.Chmod(path, 0700)
+}
```
Comment (hush-hush) [inline pkg/util/filesystem/permission_nowindows.go:22]: Comparing the Windows and linux implementation, why aren't we doing the same and changing the user/group ?

This doesn't use the current user/group. If the file is own by another user/group we would just change the right not the group or user.

File: pkg/util/filesystem/permission_windows.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	systemSid        *windows.SID
+}
+
+// Create a new instance of `Permission`
+func NewPermission() (*Permission, error) {
+	administratorSid, err := windows.StringToSid("S-1-5-32-544")
+	if err != nil {
+		return nil, err
+	}
+	systemSid, err := windows.StringToSid("S-1-5-18")
```
Comment (hush-hush) [inline pkg/util/filesystem/permission_windows.go:30]: I'm not familiar with windows: what are `S-1-5-32-544` and `S-1-5-18` ? Are they standard between all windows versions ?

Could we maybe have set in constant with a comment ? WDYT ?

File: pkg/forwarder/forwarder.go [hunk 4]
```diff
@@ -376,6 +378,13 @@ func NewDefaultForwarder(options *Options) *DefaultForwarder {
 	return f
 }
 
+func getAgentFolder(options *Options) string {
+	if HasFeature(options.EnabledFeatures, CoreFeatures) {
+		return "core"
+	}
+	return ""
+}
```
Comment (hush-hush) [inline pkg/forwarder/forwarder.go:386]: I'm guessing we would add other agent to the list later/when needed ?

File: pkg/util/filesystem/permission_windows.go (hunk 3 above)
Comment (ogaca-dd) [reply pkg/util/filesystem/permission_windows.go:30]:
```go
administratorSid, err := windows.StringToSid("S-1-5-32-544")
```
As `administratorSid` suggest, `S-1-5-32-544` is the sid of administrator. Do you think extra comment is needed?

File: pkg/forwarder/forwarder.go (hunk 4 above)
Comment (ogaca-dd) [reply pkg/forwarder/forwarder.go:386]: Yes, @truthbk  added `HasFeature` to enable features based on the Agent (core, process ...).

File: pkg/util/filesystem/permission_nowindows.go (hunk 2 above)
Comment (ogaca-dd) [reply pkg/util/filesystem/permission_nowindows.go:22]: >why aren't we doing the same and changing the user/group ?
The purpose of this method is to restrict the access of the folder to the Agent or an administrator only.

**Review by ruthnaebeck — APPROVED** [review]
👍  for docs

File: pkg/util/filesystem/permission_windows.go (hunk 3 above)
Comment (hush-hush) [reply pkg/util/filesystem/permission_windows.go:30]: I'd say if this is obvious to Windows dev we can leave it as is. If not, maybe a quick comment explaining it's an hardcoded ID for the Windows administrator and it's compatible with all windows version ?

You know more about Windows than me, I let you decide !

**Review by hush-hush — APPROVED** [review]

**Pull request merged**